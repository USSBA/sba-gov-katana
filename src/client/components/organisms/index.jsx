/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
