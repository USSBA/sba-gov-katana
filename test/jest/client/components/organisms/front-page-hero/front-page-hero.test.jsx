import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { FrontPageHero } from 'organisms'

const links = [
  {
    link: '/a',
    title: 'A'
  },
  {
    link: '/b',
    title: 'B'
  }
]

const jsx = (
  <FrontPageHero
    color="blue"
    image="image.png"
    imageAlt="some image"
    links={links}
    title="Title"
  />
)

describe('front page hero', () => {
  test('renders correctly', () => {
    const tree = renderer.create(jsx).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('callout buttons render correct event config', () => {
    const basicLinks = shallow(jsx)
      .find('BasicLink')
      .map((basicLink, index) => {
        const { title } = links[index]
        const eventConfig = {
          category: 'Front-Page-Non-Featured-Hero',
          action: `Click: ${title}`
        }
        expect(basicLink.props().eventConfig).toEqual(eventConfig)
      })
  })
})
