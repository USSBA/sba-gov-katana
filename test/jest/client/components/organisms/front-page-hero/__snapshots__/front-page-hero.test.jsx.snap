// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`front page hero renders correctly 1`] = `
<div
  className="container blue "
>
  <div
    className="titleContainer"
  >
    <p
      className="title"
    >
      Title
    </p>
  </div>
  <div
    className="imageContainer"
  >
    <img
      alt="some image"
      className="Banner"
      src="image.png"
    />
  </div>
  <div
    className="boxContainer"
  >
    <div
      className="AccentBox"
    >
      <ul>
        <a
          autoFocus={false}
          className=""
          onKeyPress={[Function]}
          onTouchTap={[Function]}
          tabIndex="0"
        >
          <li
            className="borderBox"
          >
            A
          </li>
        </a>
        <a
          autoFocus={false}
          className=""
          onKeyPress={[Function]}
          onTouchTap={[Function]}
          tabIndex="0"
        >
          <li
            className="borderBox"
          >
            B
          </li>
        </a>
      </ul>
      <img
        alt=""
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;
